python/ucxx/_lib/libucxx.pyx

```diff
@@ -40,15 +41,45 @@ include "tag.pyx"
 logger = logging.getLogger("ucx")
 
 
-np.import_array()
-
+cdef class HostBufferAdapter:
+    """A simple adapter around HostBuffer implementing the buffer protocol"""
+    cdef Py_ssize_t _size
+    cdef void* _ptr
+    cdef Py_ssize_t[1] _shape
+    cdef Py_ssize_t[1] _strides
+    cdef Py_ssize_t _itemsize
+
+    @staticmethod
+    cdef _from_host_buffer(HostBuffer* host_buffer):
+        """Construct a new HostBufferAdapter from a HostBuffer.
+
+        This factory takes ownership of the input host_buffer's data, so
+        attempting to use the input after this function is called will result
+        in undefined behavior.
+        """
+        cdef HostBufferAdapter obj = HostBufferAdapter.__new__(HostBufferAdapter)
+        obj._size = host_buffer.getSize()
+        obj._ptr = host_buffer.release()
+        obj._shape = [obj._size]
+        obj._itemsize = sizeof(uint8_t)
+        obj._strides = [obj._itemsize]
+        return obj
+
+    def __getbuffer__(self, Py_buffer *buffer, int flags):
+        buffer.buf = self._ptr
+        buffer.format = 'c'
+        buffer.internal = NULL
+        buffer.itemsize = self._itemsize
+        buffer.len = self._size * self._itemsize
+        buffer.ndim = 1
+        buffer.obj = self
```

Comment:
Fair point, we'd be at the mercy of Cython setting up appropriate handling of errors to properly decrement the ref count here otherwise you'd like. I'll reorder.